src/hotspot/os/windows/os_windows.cpp

```diff
@@ -3324,16 +3324,27 @@ static size_t large_page_init_decide_size() {
     WARN("Large page is not supported by the processor.");
     return 0;
   }
-
-#if defined(IA32) || defined(AMD64)
-  if (size > 4*M || LargePageSizeInBytes > 4*M) {
+#if defined(IA32)
+    if (size > 4 * M || LargePageSizeInBytes > 4 * M) {
     WARN("JVM cannot use large pages bigger than 4mb.");
     return 0;
   }
+#elif defined(AMD64)
+    if (!EnableAllLargePageSizesForWindows) {
+      if (size > 4 * M || LargePageSizeInBytes > 4 * M) {
+        WARN("JVM cannot use large pages bigger than 4mb.");
+        return 0;
+      }
+    }
 #endif
-
-  if (LargePageSizeInBytes > 0 && LargePageSizeInBytes % size == 0) {
-    size = LargePageSizeInBytes;
+  if (LargePageSizeInBytes > 0) {
+    if (LargePageSizeInBytes % size == 0) {
+      size = LargePageSizeInBytes;
+    } else {
+      WARN("The specified large page size (%d) is not a multiple of the minimum large page size (%d), defaulting to minimum page size.", LargePageSizeInBytes, size);
+    }
+  } else {
+    WARN("The JVM cannot use large pages because the large page size setting is not configured, defaulting to minimum page size (%d).", size);
```

Comment:
This message looks wrong; LargePageSizeInBytes=0 is documented to use the default large page size, and it seems that the surrounding code does just that. Is the warning even necessary?

Reply:
You are correct; the warning message does seem misleading. The LargePageSizeInBytes=0 is meant to use the default large page size, which the surrounding code already handles. Therefore, the warning about large pages not being configured is unnecessary. 
We can remove this warning and adjust the code to reflect the intended use of default large page sizes when 
LargePageSizeInBytes is zero.